1.Exploratory_Data_Analysis/scripts/2a.correlation_nomic_features.py

```diff
@@ -0,0 +1,233 @@
+]
+
+results = [
+    (nomic_df["Metadata_inducer1_concentration_value"].astype(str)).astype(str),
+    (
+        nomic_df["Metadata_inducer1_concentration_value"].astype(str)
+        + "_"
+        + nomic_df["Metadata_inducer2_concentration_value"].astype(str)
+    ).astype(str),
+]
+nomic_df["Metadata_Dose"] = np.select(condlist=conditions, choicelist=results)
+nomic_df["Metadata_inducer1_concentration_value"] = pd.to_numeric(
+    nomic_df["Metadata_inducer1_concentration_value"]
+)
+# one beta of inudcer1, inducer1 concentration, inhibitor, and inhibitor concentration all as 1 beta term
+nomic_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    nomic_df["Metadata_Treatment"]
+    + "_"
+    + nomic_df["Metadata_Dose"].astype(str)
+    + "_"
+    + nomic_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + nomic_df["Metadata_inhibitor_concentration_value"].astype(str)
+).astype(str)
+
+
+# In[6]:
+
+
+nomic_cleaned = nomic_df.copy()
+# drop first column of metadata
+nomic_df.columns[3:25]
+nomic_df = nomic_df.drop(nomic_df.columns[3:25], axis=1)
+nomic_df = nomic_df.drop(nomic_df.columns[0:2], axis=1)
+nomic_df.drop(nomic_df.columns[0], axis=1, inplace=True)
+# drop Metadata_Dose column
+nomic_df = nomic_df.drop(["Metadata_Dose"], axis=1)
+nomic_df = nomic_df.drop(["Metadata_Treatment"], axis=1)
+nomic_df = nomic_df.drop(["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"], axis=1)
+
+
+# # Normalization of Values
+
+# In[7]:
+
+
+# min-max normalization of nomic data from scipy
+from sklearn.preprocessing import MinMaxScaler
+
+scaler = MinMaxScaler()
+nomic_df = pd.DataFrame(scaler.fit_transform(nomic_df), columns=nomic_df.columns)
+
+
+# In[8]:
+
+
+# summary statistics of df to check min-max normalization
+nomic_df.describe()
+
+
+# In[9]:
+
+
+# add position_x back to df
+nomic_df.loc[:, "Metadata_position_x"] = nomic_df_raw["position_x"]
+
+
+# # Correlation of Wells
+
+# In[10]:
+
+
+# set index to Metadata_Well
+nomic_df = nomic_df.set_index("Metadata_position_x")
+
+
+# In[11]:
+
+
+well_corr_df = nomic_df.T.corr()
+save_path = pathlib.Path(f"./results/correlation/{cell_type}/nomic/")
+save_path.mkdir(parents=True, exist_ok=True)
+well_corr_df.to_csv(f"{save_path}/wells_corr.csv")
+
+
+# # All Treatment correlation
+
+# In[12]:
+
+
+nomic_df.reset_index(inplace=True)
+nomic_df.drop(["Metadata_position_x"], axis=1, inplace=True)
+nomic_df.loc[:, "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = nomic_cleaned[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+]
+# groupby Metadata_Treatment_Dose_Inhibitor_Dose
+nomic_df = nomic_df.groupby("oneb_Metadata_Treatment_Dose_Inhibitor_Dose").mean()
+
+
+# In[13]:
+
+
+well_corr_df = nomic_df.T.corr()
+save_path = pathlib.Path(f"./results/correlation/{cell_type}/nomic/")
+save_path.mkdir(parents=True, exist_ok=True)
+well_corr_df.to_csv(f"{save_path}/treatments_corr.csv")
+
+
+# # Treatment correlation for selected treatments
+
+# In[14]:
+
+
+list_of_treatments = [
```

Comment:
We set constants like this after package imports and function definitions. If you plan on using this list in the future, you may also consider adding it to a separate utils.py file and importing it

Reply:
addressed via adding a `params.py` file for import